proposal.rst

```diff
@@ -168,82 +173,124 @@ Project Details
 
 **Describe any existing work and concepts on which your project is based.**
 
-FILLME
+First, I have collected several resources about how to implement and improve the estimation.
+
+*** Academic
+1. (Estimating Corpus Size via Queries [CIKM'06])[http://fontoura.org/papers/cikm2006.pdf] This paper uses the basic estimator estimate the search result size. The first approach requires a uniform random sample of documents from the corpus. It avoids this notoriously difficult sample generation problem, and instead uses two fairly uncorrelated sets of terms as query pools; the accuracy of the second approach depends on the degree of correlation among the two sets of terms.
+2. (Random Sampling from a Search Engine’s Index [WWW'06]) [http://www2006.org/programme/files/pdf/3047.pdf] This paper proposes a lexicon-based technique and a random
+walk technique to avoid long documents and highly ranked documents.
```

Comment:
This paper is about a technique to sample documents randomly from a search engines index when you can only run queries.

If we wanted to sample documents randomly in Xapian we don't need to go to such lengths.  In the case where the docids are contiguous it's very easy (and we can tell if this is the case).  When there are gaps we could just pick a random docid in the range that's used and resample if we get an unused one, though maybe there's a more sophisticated approach.  But even this is going to be much more efficient than running queries to sample documents.